### PR TITLE
apply rt machine config pool

### PR DESCRIPTION
**Pull request by karmab**
# Description

This makes sure:
- worker-rt machineconfig pool is created
- documents how to tag all nodes but masters

## Type of change

Please select the appropiate options:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Testing

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] launch make deploy and wait for mcp to be applied and real time kernel to be updated on the worker nodes

**Test Configuration**:

- Versions: ocp4.3
- Hardware: any

## Checklist

- [X] I have performed a self-review of my own code
- [ ] If a change is adding a feature, it should require a change to the README.md and the review should catch this.
- [ ] If the change is a fix, it should have an issue. The review should make sure the comments state the issue (not just the number) and it should use the keywords that will close the issue on merge.
- [ ] A change should not being merged unless it passes CI or there is a comment/update saying what testing was passed.
- [ ] PRs should not be merged unless positively reviewed.

### CODE REVIEW

File: features/rt/hack/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -euo pipefail
+
+assetsDir=$(dirname "$0")/../assets
+manifestsDir=$(dirname "$0")/../manifests
+
+for node in $(oc get nodes --selector='!node-role.kubernetes.io/master' -o name); do 
+oc label $node node-role.kubernetes.io/worker-rt=""
```
Comment (e-minguez) [inline features/rt/hack/deploy.sh]: missing tab/spaces

**Comment by e-minguez** [comment]
LGTM

File: features/rt/hack/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
```
Comment (cwilkers) [inline features/rt/hack/deploy.sh]: This script needs to be executable.
```$ make deploy
./hack/deploy.sh
make: execvp: ./hack/deploy.sh: Permission denied
make: *** [Makefile:10: deploy] Error 127
```

**Comment by cwilkers** [comment]
I'm getting a node in a permanent boot loop because the `rt-kernel-patch.sh` does not actually include code to swap to the RT kernel, it only checks if the kernel is currently RT and reboots if not.

File: features/rt/hack/deploy.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+set -euo pipefail
+
+assetsDir=$(dirname "$0")/../assets
+manifestsDir=$(dirname "$0")/../manifests
+
+for node in $(oc get nodes --selector='!node-role.kubernetes.io/master' -o name); do 
+  oc label $node node-role.kubernetes.io/worker-rt=""
+done
+oc create -f $manifestsDir/00-mcp-worker-rt.yaml
+oc create -f $manifestsDir/06-mc-rtkernel-worker-rt.yaml
```
Comment (yanirq) [inline features/rt/hack/deploy.sh]: Can this be a one stop shop for all other manifests ?  (e.g. performance, cpu isolation)  if so this needs to move to an upper level.

File: features/rt/hack/deploy.sh (hunk 2 above)
Comment (karmab) [reply features/rt/hack/deploy.sh]: fixed by moving the mcp to its own feature directory

File: features/rt/hack/deploy.sh (hunk 3 above)
Comment (karmab) [reply features/rt/hack/deploy.sh]: fixed by moving the mcp to its own feature directory

File: features/mcp/00-mcp-worker-rt.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfigPool
+metadata:
+  name: worker-rt
+spec:
+  machineConfigSelector:
+    matchExpressions:
+      - {key: machineconfiguration.openshift.io/role, operator: In, values: [worker,worker-rt]}
```
Comment (cynepco3hahue) [inline features/mcp/00-mcp-worker-rt.yaml]: hm I can see you want to apply machine configs for worker and worker-rt labels, but I thought the worker machine-config-pool will apply worker machine configs, so why do we need to apply it twice?

Comment (karmab) [reply features/mcp/00-mcp-worker-rt.yaml]: i believe it's required by mcp

Comment (cynepco3hahue) [reply features/mcp/00-mcp-worker-rt.yaml]: probably because of the bug https://bugzilla.redhat.com/show_bug.cgi?id=1772490

**Pull request merged**